### ISSUE TITLE
Mumble inaccessible to blind users

### ISSUE THREAD

**Issue by robin24** [mumble-voip/mumble#1337]
Hi,
I'm a fully blind computer user using Mac OS 10.9. In order to access a computer, I use a screen-reader called VoiceOver, which reads on-screen text using synthetic speech. This software is already part of Mac OS, and can easily be enabled on any Mac by pressing Command+F5. Unfortunately, the Mumble GUI has been designed in a way that VoiceOver is unable to read any window content and will simply say 'unknown' within the Mumble program. No controls, labels, or any other UI elements are currently visible to VoiceOver, making it impossible for blind users to utilize this great VoIP software.
The main issue is most likely the QT framework. Unfortunately, QT isn't very friendly in regards to accessibility, however in recent versions this has been improved with more accessibility-related properties becoming available to developers. Please, would it be possible to investigate this further and possibly (hopefully) implement some degree of accessibility?
Many thanks!

**Comment by robin24** [comment]
Hi,
since quite a lot of time has passed since I opened this ticket, I just wanted to ask whether accessibility-related work is still in progress… I just tested the latest development snapshot, 1.3.0~511~ge469bd6~snapshot. I discovered that the initial audio assistant was mostly accessible, and I could go through that and make adjustments without too much trouble. However, once in the main window I discovered that connecting to a server is pretty much impossible, as most of that window is inaccessible… I managed to connect to a server with much trial and error, but once connected I couldn’t do anything… I think I could read 1 or 2 channel names, but that was it… The menu war items are fully accessible though, so maybe if the connection window cannot be made accessible, it might be an idea to put an option in there that brings up a simple window for the IP / hostname, port and user name, and will automatically connect to the server once those fields have been filled and submitted… The same could perhaps be done for the channels, offer a text field where a channel name can be entered which will then join that channel and allow the user to speak with other connected users… Still wouldn’t be a perfect solution, but it would at least make Mumble more usable for screen reader users.
Thanks!
Robin=

**Comment by Sophira** [comment]
This is still a big issue with the newest dev builds, unfortunately. The server connection window is still inaccessible.

I'm new to the Mumble codebase, Qt development, and OS X in general, so I'm probably not going to be much help, but I'm going to take a closer look and see what can be done. Please don't expect much though! If someone with more experience than me wants to take a look into this, please do - it's really causing issues.

**Comment by robin24** [comment]
Hi Sophira,
thanks for your response and for confirming the issues! I’ve also looked at dev builds from time to time and I’m aware of the issue with the connection window. However, unfortunately that’s not the only problem! I once received help from a sighted friend who connected me to a server, unfortunately I then found that there was pretty much nothing that was accessible afterwards. So, while an accessible connection window would certainly be a step in the right direction, this sadly wouldn’t really make Mumble more usable for us. Making Mumble more accessible would require greater commitment to this issue from the core devs, however if we look at the fact that this issue has been open for almost 2 years without any noteworthy improvements to Mumble, it is clear that this commitment is clearly lacking.
Best regards,
Robin
